app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import icons from '@/assets/icons.svg'
+import DropdownMenu from '@/components/DropdownMenu.vue'
+import SvgButton from '@/components/SvgButton.vue'
+import { ref } from 'vue'
+import { TextFormatOptions } from './visualizations/TableVisualization.vue'
+
+const emit = defineEmits<{
+  changeFormat: [formatValue: TextFormatOptions]
+}>()
+
+const textFormatterSelected = ref(TextFormatOptions.Partial)
+const setTextFormatterSelected = (option: TextFormatOptions) => {
+  textFormatterSelected.value = option
+  emit('changeFormat', textFormatterSelected.value)
+}
+
+const iconPath = `${icons}#paragraph`
+
+const open = ref(false)
+
+const toggleMenu = () => {
+  open.value = !open.value
+}
+</script>
+
+<template>
+  <DropdownMenu v-model:open="open" class="TextFormattingSelector">
+    <template #button
+      ><SvgButton name="paragraph" title="Text Display Options" @click.stop="toggleMenu()" />
+    </template>
+
+    <template #entries>
+      <button
+        :class="`${textFormatterSelected === TextFormatOptions.On && 'selected'}`"
```

Comment:
There is a Vue syntax for doing this more safely than string interpolation:
```suggestion
        :class="{ selected: textFormatterSelected === TextFormatOptions.On }"
```